File: backend/src/main/java/vn/edu/fpt/doghandbook/backend/service/HealthSessionService.java
package vn.edu.fpt.doghandbook.backend.service;

import vn.edu.fpt.doghandbook.backend.dto.request.HealthSessionRequest;
import vn.edu.fpt.doghandbook.backend.dto.request.SessionFollowUpRequest;
import vn.edu.fpt.doghandbook.backend.dto.response.HealthSessionResponse;
import vn.edu.fpt.doghandbook.backend.dto.response.PageResponse;

import java.time.LocalDateTime;

public interface HealthSessionService {

    HealthSessionResponse create(HealthSessionRequest request, Integer trainerId);

    HealthSessionResponse getById(Integer sessionId);

    PageResponse<HealthSessionResponse> getByTrainer(Integer trainerId, int page, int size);

    PageResponse<HealthSessionResponse> getByDog(Integer dogId, int page, int size);

    HealthSessionResponse addFollowUp(SessionFollowUpRequest request, Integer trainerId);

    HealthSessionResponse resolve(Integer sessionId, String resolutionNotes, Integer trainerId,
                                   LocalDateTime localUpdatedAt);
}

File: web-admin/src/services/healthSessionService.js
import api from './api';

export const healthSessionService = {
  getByDog: (dogId, page = 0, size = 10) =>
    api.get(`/health-sessions/by-dog/${dogId}?page=${page}&size=${size}`),
  getById: (id) => api.get(`/health-sessions/${id}`),
  create: (data) => api.post('/health-sessions', data),
  addFollowUp: (id, data) => api.post(`/health-sessions/${id}/follow-up`, data),
  resolve: (id, data) => api.put(`/health-sessions/${id}/resolve`, data),
};
